### PR TITLE
Refactor Content Mocks & Add Image Blob Mocks

### PR DESCRIPTION
**Pull request by mmichaelis**
Previous content mocks used some sophisticated number magic to get a content of a given behavior by a magic id.

The new approach is not using magic, but pure configuration to get a special behavior on contents.

Services like the `MockContentDisplayService` now use this new mocking behavior.

The desired goal is to provide an easier to use and easier to extend API for mocking.

The new mocking approach now also supports (Image) Blobs, so that we can test recent enhancements towards displaying embedded images in CoreMedia RichText.

### CODE REVIEW

File: packages/ckeditor5-coremedia-studio-integration-mock/src/content/PredefinedMockContents.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    .filter((c) => (c.blob?.length || 0) === 0)
+    // Get some useful name and the URI Path for the link.
+    .map((c) => {
+      const { id, name: mockName, type, comment } = c;
+      const uriPath = contentUriPath(id);
+      let name: string;
+      if (!!comment) {
+        name = comment;
+      } else if (Array.isArray(mockName)) {
+        name = `${capitalize(type)} with name toggle`;
```
Comment (pkliesch) [inline packages/ckeditor5-coremedia-studio-integration-mock/src/content/PredefinedMockContents.ts:201]: type is possibly undefined

File: packages/ckeditor5-coremedia-studio-integration-mock/src/content/ObservableMutableProperty.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  readonly #valuesLength: number;
+  /**
+   * Signals, if there is any value present (thus, `true` implies `valueLength > 0`).
+   */
+  readonly #anyValue: boolean;
+  /**
+   * Signals, if there is more than one value present, i.e., if we shall loop
+   * or not. Implies `valueLength > 1`.
+   */
+  readonly #iterating: boolean;
```
Comment (pkliesch) [inline packages/ckeditor5-coremedia-studio-integration-mock/src/content/ObservableMutableProperty.ts:39]: this property seems to be unused

File: packages/ckeditor5-coremedia-studio-integration-mock/src/content/PredefinedMockContents.ts (hunk 1 above)
Comment (mmichaelis) [reply packages/ckeditor5-coremedia-studio-integration-mock/src/content/PredefinedMockContents.ts:268]: Meanwhile a fallback is used:

https://github.com/CoreMedia/ckeditor-plugins/blob/557c4bee6b132f05d271dc574f34214197ebf3f5/packages/ckeditor5-coremedia-studio-integration-mock/src/content/PredefinedMockContents.ts#L262

File: packages/ckeditor5-coremedia-studio-integration-mock/src/content/ObservableMutableProperty.ts (hunk 2 above)
Comment (mmichaelis) [reply packages/ckeditor5-coremedia-studio-integration-mock/src/content/ObservableMutableProperty.ts:39]: Removed (at some point in time 😄).

**Pull request merged**